### ISSUE TITLE
Bogus NoTrailingSpaces findings in some files

### ISSUE THREAD

**Issue by sschuberth** [arturbosch/detekt#1688]
## Expected Behavior

`NoTrailingSpaces` is only reported for lines with trailing whitespace.

## Observed Behavior

`NoTrailingSpaces` is sometimes reported for lines without trailing whitespace.

## Steps to Reproduce

1. Checkout https://github.com/heremaps/oss-review-toolkit (commit 45cfcb53911634e518f2ed7ced1f9743fc57fec4)
2. Run `./gradlew :analyzer:detekt`
3. Observe the bogus report of `NoTrailingSpaces - [PsiWhiteSpace] at C:\Dev\ort\analyzer\src\main\kotlin\managers\GoDep.kt:1:1`

## Context

Note that, if a bogus finding is reported, I so far always saw it being reported at `1:1`.

## Your Environment
* Version of detekt used: 1.0.0-RC15
* Version of Gradle used (if applicable): 4.5.1
* Operating System and version: Windows 10
* Link to your project (if it's a public repository): https://github.com/heremaps/oss-review-toolkit

**Comment by cortinico** [comment]
Hey @sschuberth 
Apparently the `NoTrailingSpaces` is a KtLint rule (here https://ktlint.github.io/#rule-trailing-whitespaces).
Can I ask you to check if this issue is reported also using KtLint standalone. If so, you should probably open an issue on their repo.

**Comment by sschuberth** [comment]
I've tried with stand-alone *ktlint* 0.32.0 and 0.33.0, and neither reports a bogus whitespace at `1:1`.

**Comment by cortinico** [comment]
Hey @sschuberth 
I checked out your repo locally and tried to run `./gradlew :analyzer:detekt`. Apparently there is no report for `GoDep.kt:1:1` (although there are others reports at `1:1`). Here the result of detekt:

```
> Task :analyzer:detekt
........................................................

56 kotlin files were analyzed.

Ruleset: comments
Ruleset: complexity - 5h 40min debt
   LargeClass - 198/150 - [DotNetSupport] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:44:1
        NestedBlockDepth - 5/5 - [getPackageReferenceFromRestAPI] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:187:5
        LargeClass - 218/150 - [Stack] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Stack.kt:55:1
        ComplexMethod - 20/20 - [parseKeyValue] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Stack.kt:237:5
        NestedBlockDepth - 5/5 - [parseKeyValue] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Stack.kt:237:5
        LongMethod - 157/80 - [resolveDependencies] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Pip.kt:209:5
        LargeClass - 302/150 - [Pip] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Pip.kt:133:1
        ComplexMethod - 32/20 - [resolveDependencies] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Pip.kt:209:5
        LargeClass - 155/150 - [Bundler] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Bundler.kt:62:1
        LongMethod - 117/80 - [parseInstalledModules] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Npm.kt:177:5
        LargeClass - 302/150 - [Npm] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Npm.kt:67:1
        ComplexMethod - 21/20 - [parseInstalledModules] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Npm.kt:177:5
        NestedBlockDepth - 9/5 - [parseInstalledModules] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Npm.kt:177:5
        LargeClass - 197/150 - [PhpComposer] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/PhpComposer.kt:61:1
        ComplexCondition - 4/4 - [<anonymous>] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/PhpComposer.kt:178:17
        LongMethod - 89/80 - [requestRemoteArtifact] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/MavenSupport.kt:297:5
        LargeClass - 251/150 - [MavenSupport] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/MavenSupport.kt:89:1
Ruleset: empty-blocks
Ruleset: exceptions - 2h 20min debt
   TooGenericExceptionCaught - [e] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:154:18
        TooGenericExceptionCaught - [e] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:168:18
        TooGenericExceptionCaught - [e] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:244:18
        TooGenericExceptionThrown - [informationUrl] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:152:18
        TooGenericExceptionThrown - [packageJsonNode] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/DotNetSupport.kt:243:20
        TooGenericExceptionCaught - [e] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/PackageManager.kt:212:26
        TooGenericExceptionCaught - [e] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/MavenSupport.kt:397:26
Ruleset: formatting - 30min debt
   NoTrailingSpaces - [PsiWhiteSpace] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Bower.kt:1:1
        ChainWrapping - [PsiElement(ANDAND)] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/Bower.kt:163:21
        NoConsecutiveBlankLines - [PsiWhiteSpace] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/DotNet.kt:123:2
        NoMultipleSpaces - [PsiWhiteSpace] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/GoDep.kt:60:24
        NoTrailingSpaces - [PsiWhiteSpace] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/PhpComposer.kt:1:1
        ChainWrapping - [PsiElement(ANDAND)] at /Users/nco/playground/oss-review-toolkit/analyzer/src/main/kotlin/managers/PhpComposer.kt:179:17
Ruleset: naming
Ruleset: performance
Ruleset: potential-bugs
Ruleset: style

Overall debt: 8h 30min
```

I see there is a report for `Bower.kt:1:1`. Is that the problem that you're referring to?

**Comment by sschuberth** [comment]
> Is that the problem that you're referring to?

Yes. So it seems to be somewhat non-deterministic WRT the file(s) in which the bogus issue is reported. The finding in `PhpComposer.kt:1:1` is also bogus.

**Comment by arturbosch** [comment]
Thanks I could reproduce and fix this.

**Comment by sschuberth** [comment]
FYI, I'm seeing this (again) in detekt 1.0.0. What I got was:

```
   NoMultipleSpaces - [PsiWhiteSpace] at oss-review-toolkit/analyzer/src/main/kotlin/managers/GoDep.kt:158:27
        NoTrailingSpaces - [PsiWhiteSpace] at oss-review-toolkit/analyzer/src/main/kotlin/managers/GoDep.kt:1:1
```
But once I fixed the `NoMultipleSpaces` in `158:27`, the bogus `NoTrailingSpaces` reported at `1:1` also disappeared.
